src/Java.Interop.Tools.JavaTypeSystem/JavaModels/JavaParameterModel.cs

```diff
@@ -0,0 +1,56 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Java.Interop.Tools.JavaTypeSystem.Models
+{
+	public class JavaParameterModel : IJavaResolvable
+	{
+		public string Name { get; }
+		public string Type { get; }
+		public string JniType { get; }
+		public bool IsNotNull { get; }
+		public string GenericType { get; }
+
+		public JavaMethodModel ParentMethod { get; }
+		public JavaTypeReference? TypeModel { get; private set; }
```

Comment:
Not sure what this means?

Reply:
The `TypeModel` is the resolved `JavaTypeReference` model for the `Type` string.

Similar terminology is used elsewhere, ie: `JavaMethodModel.ReturnTypeModel` is the resolved model for `JavaMethodModel.ReturnType` string.